### PR TITLE
docs: Update Config Validator with new validation result

### LINKED ISSUE #1175: Update CI Integrations Config Validator with new validation result
**Description**

Introducing a CI Integrations Config Validator provides an ability to validate the CI Integrations configuration fields by running the validate command that provides information about the possible errors in the configuration file.
For example, if the configuration file contains a non-valid email/password used to log in into CI, the user will get the corresponding error message once run the validate command.

**Acceptance Criteria**
- [ ] update [CI Integrations Config Validator
](https://github.com/platform-platform/monorepo/blob/master/metrics/ci_integrations/docs/04_ci_integrations_config_validator.md) with suggested changes from https://gist.github.com/solid-yevheniiashchypeleva/cebfddfc7f309df22dded23a10a24c5c

### PR DESCRIPTION
**Pull request by solid-yevheniiashchypeleva**
Fixes #1175

## Checklist

- [x] Documented

### CODE REVIEW

File: metrics/ci_integrations/docs/diagrams/config_validator_destination_class_diagram.puml [hunk 1]
```diff
@@ -56,13 +51,17 @@
                     + results: UnmodifiableMapView<ConfigField, FieldValidationResult>
                 }
 
-                class FieldValidationResult {
+                class FieldValidationResult<T> {
```
Comment (solid-vovabeloded) [inline metrics/ci_integrations/docs/diagrams/config_validator_destination_class_diagram.puml:54]: It seems like we've missed the relation between this class and `CoolIntegrationDestinationValidationDelegate` class.

File: metrics/ci_integrations/docs/diagrams/config_validator_sequence_diagram.puml [hunk 2]
```diff
@@ -54,14 +54,14 @@ alt auth is valid
 
     CoolIntegrationClient --> CoolIntegrationDestinationValidationDelegate : InteractionResult.success
     CoolIntegrationDestinationValidationDelegate -> CoolIntegrationDestinationValidationDelegate : performs additional validation
-    CoolIntegrationDestinationValidationDelegate --> CoolIntegrationDestinationValidator : InteractionResult.success
-    CoolIntegrationDestinationValidator -> ValidationResultBuilder : setResult(CoolIntegrationConfigField.auth, FieldValidationResult.success())
+    CoolIntegrationDestinationValidationDelegate --> CoolIntegrationDestinationValidator : authFieldValidationResult
```
Comment (solid-vovabeloded) [inline metrics/ci_integrations/docs/diagrams/config_validator_sequence_diagram.puml:57]:
```suggestion
    CoolIntegrationDestinationValidationDelegate --> CoolIntegrationDestinationValidator : FieldValidationResult.success
```
Would it be a bit more clear? If so, please, apply this suggestion to all similar situations.

**Review by solid-vovabeloded — CHANGES_REQUESTED** [review]
Great job, @solid-daniyilyevtyushkin & @solid-yevheniiashchypeleva! I've added a couple of minor suggestions - please take a look. Overall, the document updates look good, thanks!

**Review by solid-vovabeloded — APPROVED** [review]
LGTM, thanks @solid-daniyilyevtyushkin & @solid-yevheniiashchypeleva!

**Comment by solid-daniyilyevtyushkin** [comment]
@Mergifyio refresh

**Pull request merged**